packages/edit-site/src/components/sidebar-navigation-screen-page/page-details.js

```diff
@@ -0,0 +1,163 @@
+	}
+
+	const details = [
+		{
+			label: __( 'Status' ),
+			value: (
+				<StatusLabel
+					status={ page?.password ? 'protected' : page.status }
+					date={ page?.date }
+				/>
+			),
+		},
+		{
+			label: __( 'Slug' ),
+			value: <Truncate numberOfLines={ 1 }>{ page.slug }</Truncate>,
+		},
+	];
+
+	if ( page?.templateTitle ) {
+		details.push( {
+			label: __( 'Template' ),
+			value: decodeEntities( page.templateTitle ),
+		} );
+	}
+
+	details.push( {
+		label: __( 'Parent' ),
+		// `null` indicates no parent.
+		value:
+			null === page?.parentTitle
+				? __( 'Top level' )
+				: decodeEntities( page?.parentTitle || __( '(no title)' ) ),
+	} );
+
+	/*
+	 * translators: If your word count is based on single characters (e.g. East Asian characters),
+	 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
+	 * Do not translate into your own language.
+	 */
+	const wordCountType = _x( 'words', 'Word count type. Do not translate!' );
+	const wordsCounted = page?.content?.rendered
+		? wordCount( page.content.rendered, wordCountType )
+		: 0;
+	const readingTime = Math.round( wordsCounted / AVERAGE_READING_RATE );
+
+	if ( wordsCounted ) {
+		details.push(
+			{
+				label: __( 'Words' ),
+				value: wordsCounted.toLocaleString() || __( 'Unknown' ),
+			},
+			{
+				label: __( 'Time to read' ),
+				value:
+					readingTime > 1
+						? sprintf(
+								/* translators: %s: is the number of minutes. */
+								__( '%s mins' ),
+								readingTime.toLocaleString()
+						  )
+						: __( '< 1 min' ),
+			}
+		);
+	}
+	return details;
+}
+
+export default function PageDetails( { id } ) {
+	const { record } = useEntityRecord( 'postType', 'page', id );
+
+	const { parentTitle, templateTitle } = useSelect(
+		( select ) => {
+			const { getEditedPostContext, getSettings } = unlock(
+				select( editSiteStore )
+			);
+			const defaultTemplateTypes = getSettings()?.defaultTemplateTypes;
+			const postContext = getEditedPostContext();
+
+			// Template title.
+			const templateSlug =
+				// Checks that the post type matches the current theme's post type, otherwise
+				// the templateSlug returns 'home'.
+				postContext?.postType === 'page'
+					? postContext?.templateSlug
+					: null;
+			const _templateTitle =
+				defaultTemplateTypes && templateSlug
+					? defaultTemplateTypes.find(
+							( template ) => template.slug === templateSlug
+					  )?.title
+					: null;
+
+			// Parent page title.
+			const _parentTitle = record?.parent
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'page',
+						record.parent,
+						{
+							_fields: [ 'title' ],
+						}
+				  )?.title?.rendered
+				: null;
+
+			return {
+				parentTitle: _parentTitle,
+				templateTitle: _templateTitle,
+			};
+		},
+		[ record ]
+	);
+	return (
+		<VStack spacing={ 5 }>
+			{ getPageDetails( {
+				parentTitle,
+				templateTitle,
+				...record,
+			} ).map( ( { label, value } ) => (
```

Comment:
Not a blocker for this PR, as I think this PR already rearranges things quite nicely, but I was wondering if having a separate `getPageDetails` function is necessary if we have a `PageDetails` component?

Another way to write it could be to have some sub-components like `DetailRow` and `DetailLabel` and `DetailValue` (still in the same file), and then instead of iterating over the results of the imperative `getPageDetails`, we could compose the structure of the page details directly with React components?

Sort of like:

```

<DetailsRow>
  <DetailsLabel>{ __( 'Status' ) }</DetailsLabel>
  <DetailsValue>...</DetailsValue>
</DetailsRow>
{ !! page?.templateTitle && (
  <DetailsRow>
    <DetailsLabel>{ __( 'Template' ) }</DetailsLabel>
    <DetailsValue>{ decodeEntities( page.templateTitle ) }</DetailsValue>
  </DetailsRow>
) }
...

```

I wouldn't worry about it at all for this PR, and it quite possibly comes down to personal preference — personally, I find reading the React-ey output a little easier to follow than reading the constructed array in `getPageDetails`, but this is pretty nit-picky 😄

Reply:
Thanks for the review, and for thinking about how to refactor the details. 

Yeah, now that you've written it out that way, I tend to agree it's easier to grok where there are components.

I'm not sure yet, but I believe that templates/template parts _may_ get the same details treatment as pages. If that occurs then there might be an obvious path for optimization according to the way you've described.

Reply:
> I'm not sure yet, but I believe that templates/template parts may get the same details treatment as pages. If that occurs then there might be an obvious path for optimization according to the way you've described.

Absolutely, I think it's perfectly fine to wait until there's a clearer path for subsequent usage 👍